### ISSUE TITLE
icons in the libnotify notification bubble

### ISSUE THREAD

**Issue by gothmog123** [andrew-bibb/cmst#157]
I know it's a minor thing but it would be great if they could follow the system icon theme.

**Comment by andrew-bibb** [comment]
Well it is supposed to work that way, but I just tested with the Oxygen and Adwaita icon themes and it does not.  Back to you in a bit (hopefully won't take long).

Thanks for reporting it.
A-

**Comment by gothmog123** [comment]
Thanks!

**Comment by andrew-bibb** [comment]
I think I've got it, new code uploaded here.  Let me know if it does not work, but it is working here.

**Comment by gothmog123** [comment]
It's working great, thanks. Icon's a little pixelated in the bubble, maybe something to do with my theme.

**Comment by andrew-bibb** [comment]
Think I know what that problem was and it is not fixed.  I was looking for icons in this order: 24x24,22x22,16,16.  Wasn't working (is now) so the icon size being selected was basically random. 

That is the good news, bad news is I'm pulling the notification code apart.  I'm sending notifications at connection state changes, which is correct except a connection state change does not care how you are connected. I had been checking the service order to present the proper icon (wired or wifi), but just discovered that does not work for the move before or move after commands.  I'm reopening until I get that hashed out.

**Comment by gothmog123** [comment]
Awesome, keep up the great work.

**Comment by andrew-bibb** [comment]
I think we are mostly there now, code here on GitHub.  Works a bit differently than before.  I'm still sending notifications at state change,but not all state changes and I'm not dealing with which particular service or technology has the online state. As part of this I've also revised the icon def. file, so everyone will get a message popup about the new def file first time you run it.  The icon definitions I changed actually were defined but not being used, so really no change from that end.

I'm leaving this open for a bit because I've discovered some oddities with the Oxygen theme, specifically there seem to be a bunch of network related Freedesktop standard icons that they don't provide.  I've dealt with some of that in the revised icon def file, but I need to go through with a fine tooth comb to see if they indeed don't have them or if maybe they are just missing from my local installation.  One of the reasons this took me so long as I was getting inconsistent icons, sometimes Oxygen, sometimes the internal ones.  Took me forever to figure out that some did not exist in Oxygen (or at least my install of Oxygen).

**Comment by gothmog123** [comment]
Maybe my icon theme is weird, or non-standard, but I'm not getting any icons now.

Maybe some problems are being caused by the fact that cmst is always looking for symbolic icons, or is that just for the tray? Those are not the ones to be used, at least in my theme.

**Comment by andrew-bibb** [comment]
Which icon theme are you using, and what are you doing to try and have a notification popup?  It is easier if I can replicate all this stuff and play with that setup.  In my day job I'm an engineer so I learn best by breaking things and seeing how they failed.  

I have a suspicion on the icons not showing up, and if I'm right it is just a matter of adding a few names to the icon.def file.  It would be a whole lot easier if people who make icon themes would just follow the freedesktop standard naming specification.

**Comment by gothmog123** [comment]
I'm using Papirus Dark GTK (from the Arch AUR) and lxqt-notifications. To get popups I restart connman for example.

Other network managers are not using the symbolic icons from my theme.

**Comment by andrew-bibb** [comment]
Well you do have interesting questions.  Installed the Papirus Dark GTK theme and same here.  The QIcon class is not returning anything for that theme, an absolutely empty stringlist.  Not sure if it is because they are .SVG icons or not.  I've added the qt5-svg package, rebuilt and the same.  This may take a bit.

**Comment by gothmog123** [comment]
Working confirmed, thanks. Sorry for the crazy theme.

**Comment by andrew-bibb** [comment]
Glad it is working.  The whole notifications code now makes a lot more sense and the icon manager actually works as I intended - all good. Thanks for sticking with it through all my mistakes.

I'm still going to go through the various icon themes (including yours) making sure I've listed names for all the icons I want to use.  That is a separate issue so  I'm going to close this one out as it seems done.
